### PR TITLE
test: add boundary case tests for search history retrieval

### PR DESCRIPTION
**Pull request by FlyJoanne**
No linked issue (test coverage enhancement)

### Summary of changes
This PR adds two small but meaningful test cases to `GetLastSearchHistoryTest.java`:

- `getMoreThanAvailableHistory`: ensures that requesting more entries than the list contains returns all available entries
- `getHistoryWhenEmpty`: verifies that an empty `StateManager` returns an empty list for both full and partial retrieval methods

These tests improve robustness against boundary cases in search history logic.

### Mandatory checks
- [x] I own the copyright of the code submitted and I license it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [/] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if change is visible to the user)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [/] Screenshots added in PR description (if change is visible to the user)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): No update needed
- [x] [Checked documentation](https://docs.jabref.org/): No update needed

### Collaborators
[@lydia-yan](https://github.com/lydia-yan) [@yoasaaa](https://github.com/yoasaaa) [@brandon-lau0](https://github.com/brandon-lau0) [@FlyJoanne](https://github.com/FlyJoanne)

### CODE REVIEW

File: jabgui/src/test/java/org/jabref/gui/search/GetLastSearchHistoryTest.java [hunk 1]
```diff
@@ -60,4 +60,25 @@ void clearSearchHistory() {
...
         assertEquals(expected, lastSearchHistory);
     }
+
+    @Test
+    void getMoreThanAvailableHistory() {
+        StateManager stateManager = new StateManager();
+        stateManager.addSearchHistory("test1");
+        stateManager.addSearchHistory("test2");
+        List<String> lastSearchHistory = stateManager.getLastSearchHistory(5);
+        List<String> expected = List.of("test1", "test2");
```
[the remark this answers is not in the document]
Comment (FlyJoanne) [reply jabgui/src/test/java/org/jabref/gui/search/GetLastSearchHistoryTest.java:70]: Thanks! For consistency, I followed the naming convention in existing tests like `duplicateSearchHistory`, which also uses `expected`.

**Comment by FlyJoanne** [comment]
The failed "Determine issue number" check is expected — this PR is a pure test coverage enhancement and is not tied to any specific issue.

**Comment by koppor** [comment]
@FlyJoanne Is this kind of a university project? Can you share the name of the course so that we can count it for our non-profit organization?

---

Other than that, the test cases are trivial, There are tests from the year 2023 - and some more complex things were implemented. https://github.com/JabRef/jabref/pull/9647#issuecomment-1466390063

**Comment by koppor** [comment]
If you need a real food for thought, then think how the new plain-CLI functionality of JabRef ["jabkit"](https://github.com/JabRef/jabref/tree/main/jabkit) can be tested. I have  [clitest](https://github.com/aureliojargas/clitest) in my notes.

**Comment by koppor** [comment]
@FlyJoanne You can also investigate https://github.com/JabRef/jabref/issues/12947

**Comment by FlyJoanne** [comment]
> @FlyJoanne You can also investigate #12947

Thank you so much for the suggestions, @koppor  

At the moment, I’ve submitted [PR #13119](https://github.com/JabRef/jabref/pull/13119) to improve test coverage of the StateManager by verifying that search history is shared and ordered correctly across multiple database contexts.

Really appreciate your feedback and guidance! 😊

**Comment by hexhacker-vee** [comment]
I'm working on this issue. I've created comprehensive boundary case tests for search history retrieval functionality.

**Progress:**
- Added 27 new boundary case tests covering all edge cases
- Tests cover getLastSearchHistory, addSearchHistory, clearSearchHistory, and getWholeSearchHistory methods
- All tests pass successfully
- Created PR #14097 with the implementation

**Test Coverage:**
- Zero/negative/large size requests
- Null/empty/whitespace string handling
- Unicode and special character support
- Cross-database behavior
- Performance testing with large datasets
- Duplicate handling and edge cases

The tests ensure robust handling of boundary conditions and edge cases that could occur in real-world usage scenarios.
